File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        
        HashSet<Integer> uniqueNumbers = new HashSet<>();

        for (int num : nums) {
            if (uniqueNumbers.contains(num)) {
                return true;
            } 
            uniqueNumbers.add(num);
        }

        return false;
    }
}

File: 217-contains-duplicate/contains-duplicate.kt
class Solution {
    fun containsDuplicate(nums: IntArray): Boolean {
        

        val seen = mutableSetOf<Int>()

        for (num in nums) {

            if (seen.contains(num)){
                return true
            }

            seen.add(num)
        }
        
        return false
    }
}